### PR TITLE
feat: add fuzzy_finder_mappings to customize mappings for filter popup in fuzzy_finder_mode

### PR DESCRIPTION
**Pull request by adoyle-h**
related issue: https://github.com/nvim-neo-tree/neo-tree.nvim/issues/451#issuecomment-1475284693

### CODE REVIEW

**Comment by cseickel** [comment]
@adoyle-h While this certainly works, it seems a little rushed. For one, it doesn't make sense for the fuzzy finder mappings config to be at the root level of the config when it only applies to the filesystem source. Also, you are exposing a lot of internal implementation details with the arguments being passed and your example.

It would probably be better to have a mappings config for the filter popup that is similar to the mappings used on the main window. In this particular example, the logic for how the focus is moved up or down should not be exposed to the end user. We should be giving them ready made command names that they just need to provide the mapping(s) for.

**Comment by adoyle-h** [comment]
> It would probably be better to have a mappings config for the filter popup that is similar to the mappings used on the main window. 

Agree. I will fix it.

> We should be giving them ready made command names that they just need to provide the mapping(s) for.

How to create mappings and bind them with the filter popup? The `input` node is created in show_filter function. Its `bufnr` is unpredictable.

File: README.md [hunk 1]
```diff
@@ -292,7 +292,14 @@ use {
               ["<c-x>"] = "clear_filter",
               ["[g"] = "prev_git_modified",
               ["]g"] = "next_git_modified",
-            }
+            },
+
```
Comment (cseickel) [inline README.md:296]:
```suggestion

```
Remove extra blank line.

**Review by cseickel — APPROVED** [review]
Looks good, thanks!

**Review by cseickel — APPROVED** [review]

**Comment by ditsuke** [comment]
Works very well, thank you!

**Pull request merged**